src/Types/PlutusAddress.purs

```diff
@@ -0,0 +1,192 @@
+module Types.PlutusAddress where
+
+import Prelude
+import Data.Maybe (Maybe(..))
+import Data.Array
+import Data.Tuple (Tuple(Tuple), fst)
+import Data.Generic.Rep (class Generic)
+import Data.Show.Generic (genericShow)
+import Data.Newtype (class Newtype, wrap)
+import Data.UInt (UInt, fromInt, toInt, (.&.), and, zshr, shl)
+import Types.ByteArray (ByteArray, byteArrayToIntArray, byteArrayFromIntArray)
+import Types.UnbalancedTransaction (PubKeyHash)
+import Types.Scripts (ValidatorHash)
+import Types.PlutusData (PlutusData(Constr, Map, List, Integer, Bytes))
+import Serialization.Address (Pointer, addressBytes)
+import Serialization.Address (Address) as Serialization
+import Serialization.Hash (ed25519KeyHashFromBytes, scriptHashFromBytes) as Hash
+import ToData (class ToData, toData)
+import FromData (class FromData, fromData)
+
+--------------------------------------------------------------------------------
+-- Credential
+--------------------------------------------------------------------------------
+
+data Credential
```

Comment:
If this type is directly based on a Plutus type, could you include a note on which revision you're working off?